eth/stagedsync/stage_execute.go

```diff
@@ -197,7 +197,15 @@ func unwindExec3(u *UnwindState, s *StageState, txc wrap.TxContainer, ctx contex
 			}
 		}
 	}
-	if err := rs.Unwind(ctx, txc.Tx, u.UnwindPoint, txNum, accumulator, changeset); err != nil {
+
+	if changeset == nil {
+		if u.CurrentBlockNumber == u.UnwindPoint {
```

Comment:
I think we may want to expand this to cover the case where `c.UnwindPoint > c.CurrentBlockNumber` . This could happen if the developer set `u.UnwindPoint` but forgot to set `u.CurrentBlockNumber`.

Reply:
done